src/agent/block_store_services/block_store_base.js

```diff
@@ -326,11 +363,7 @@ class BlockStoreBase {
     }
 
     _get_block_internal_dir(block_id) {
-        let hex_str_regex = /^[0-9a-fA-f]+$/;
-        let internal_dir = hex_str_regex.test(block_id) ?
-            block_id.substring(block_id.length - 3) + '.blocks' :
-            'other.blocks';
-        return internal_dir;
+        return get_block_internal_dir(block_id);
```

Comment:
how about getting rid of `this._get_block_internal_dir()` and replace any occurrences with `block_store_base.get_block_internal_dir()`